### ISSUE TITLE
publisher.send throw uncaught exception in case of large payload in first call

### ISSUE THREAD

**Issue by VasilSvirid** [cody-greene/node-rabbitmq-client#38]
Connection and publisher taken from sample. First .send call

```
connection = new Connection( process.env.RABBIT_MQ_URL )

publisher = connection.createPublisher( {
	confirm     : true,
	maxAttempts : 2,
	exchanges   : [ { exchange: 'NAME', type: 'topic', durable: true } ]
} )

try {
	await publisher.send( 
		{ 
			exchange   : 'NAME', 
			routingKey : 'files.process', 
				headers    : {
					refresh_token,
					ws // too large (large object instead scalar value)
			}
		}, 
		{ uuid } 
	)	
}
catch ( e ) {
	// here receive 1 exception
	// code: 'FRAME_ERROR'
	// name: 'AMQPConnectionError'
	// message: 'undefined: FRAME_ERROR - type 2, all octets = <<>>: {frame_too_large,7477,4088}'
	logger.error( ( e as any ).message )	
}


process.on( 'uncaughtException', ( err, origin ) => {
	// here receive 2 exceptions but this is unwanted exceptions
	// code: 'FRAME_ERROR'
	// name: 'AMQPConnectionError'
	// message: 'undefined: FRAME_ERROR - type 2, all octets = <<>>: {frame_too_large,7477,4088}'	
	console.error( `err '${ err }', occurred in ${ origin }` )
} )
```

CURRENT
payload large, receive 2 uncaughtException, and 1 exception in catch
// code: 'FRAME_ERROR'
// name: 'AMQPConnectionError'
// message: 'undefined: FRAME_ERROR - type 2, all octets = <<>>: {frame_too_large,7477,4088}'	

EXPECTED:
payload large, receive 1 exception in catch
// code: 'FRAME_ERROR'
// name: 'AMQPConnectionError'
// message: 'undefined: FRAME_ERROR - type 2, all octets = <<>>: {frame_too_large,7477,4088}'

**Comment by cody-greene** [comment]
This type of error will cause the connection to reset so it is also emitted by the connection object.
To catch these errors you can attach a handler like this:
```javascript
connection.on('error', (err) => {
  console.log('rabbit connection error', err)
})
```

This is documented in the quickstart section of the readme, as well as [the Connection section of the full API docs](https://cody-greene.github.io/node-rabbitmq-client/latest/classes/Connection.html):
> This represents a single connection to a RabbitMQ server (or cluster). Once created, it will immediately attempt to establish a connection. When the connection is lost, for whatever reason, it will reconnect. **This implements the EventEmitter interface and may emit error events.**

Additionally, two events are emitted, one for each publish attempt (see `maxAttempts : 2`).

You can also allow larger frame sizes with the [frameMax connection option](https://cody-greene.github.io/node-rabbitmq-client/latest/classes/Connection.html), but it sounds like your large frame was not intended.

**Comment by VasilSvirid** [comment]
You are right about on("error"). I know about it but never think ... Thank you.
Close plz

**Comment by cody-greene** [comment]
Actually I think this case could be handled a little better. I should check the size of the header frame before sending it. This would prevent a connection-level error, but the publisher channel may still need to reset. I'll keep this open until I can take a look.

**Comment by VasilSvirid** [comment]
it physically impossible, I send 2 ids + token. Less then 1 KB
And other exceptions (not connection) should be handled by catch

**Comment by cody-greene** [comment]
Fixed by #41
